### PR TITLE
Hmwk3

### PR DESCRIPTION
**Pull request by vadimkononenko**

### CODE REVIEW

File: src/itea/hmwk3/FlowerStore.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        for (int i = 0; i < sum; i++) {
+            if (i < countRose) {
+                bunchOfFlowers.add(new Rose());
+            } else if (i >= countRose && i < countChamomile + countRose) {
+                bunchOfFlowers.add(new Chamomile());
+            } else {
+                bunchOfFlowers.add(new Tulip());
+            }
+        }
+
```
Comment (Aleksandr04) [inline src/itea/hmwk3/FlowerStore.java]: remove empty line

File: src/itea/hmwk3/FlowerStore.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public List<Flowers> sellSequence(int countRose, int countChamomile, int countTulip) {
+        List<Flowers> bunchOfFlowers = new ArrayList<>(countRose+countChamomile+countTulip);
+        int max = Math.max(countRose,countChamomile);
+        max = Math.max(max,countTulip);
+
+        for(int i = 0; i < max; i++){
+            if(i < countRose){
+                bunchOfFlowers.add(new Rose());
+            }else if(i < countChamomile){
```
Comment (Aleksandr04) [inline src/itea/hmwk3/FlowerStore.java]: ctrl+alt+l

File: src/itea/hmwk3/Flowers.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package itea.hmwk3;
+
+public class Flowers {
```
Comment (Aleksandr04) [inline src/itea/hmwk3/Flowers.java]: empty class?

File: src/itea/hmwk3/Runner.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package itea.hmwk3;
+
+public class Runner {
```
Comment (Aleksandr04) [inline src/itea/hmwk3/Runner.java]: empty class?

File: src/itea/lessons/Runner.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package itea.lessons;
+
+public class Runner {
```
Comment (Aleksandr04) [inline src/itea/lessons/Runner.java]: empty class?

File: src/itea/hmwk3/Chamomile.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package itea.hmwk3;
```
Comment (Aleksandr04) [inline src/itea/hmwk3/Chamomile.java]: bad package name

File: src/itea/hmwk3/FlowerStore.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                        .add(new Tulip());
+            }
+        }
+        moneyInShop(bunchOfFlowers);
+
+        return bunchOfFlowers;
+    }
+
+    public void moneyInShop(List<Flowers> flowers) {
+        for (Flowers f : flowers) {
```
Comment (Aleksandr04) [inline src/itea/hmwk3/FlowerStore.java]: refactor it on Stream

File: src/itea/hmwk3/FlowersSaver.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class FlowersSaver {
+    public static void save(String path, List<Flowers> flowers) {
+        try {
+            FileWriter fw = new FileWriter(path);
+            BufferedWriter bw = new BufferedWriter(fw);
+            for (Flowers f : flowers) {
```
Comment (Aleksandr04) [inline src/itea/hmwk3/FlowersSaver.java]: refactor it on Stream

File: src/itea/hmwk3/FlowersSaver.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+                bw.write(f.toString());
+            }
+            bw.flush();
+            bw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (Aleksandr04) [inline src/itea/hmwk3/FlowersSaver.java]: use sout("error message");

File: src/itea/hmwk3/Runner.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        List<Flowers> flowersF = fs.sell(2,4,5);
+
+        for(Flowers f : flowersF){
+            System.out.print(f + ",");
+        }
+
+        System.out.println();
+
+        List<Flowers> flowersS = fs.sellSequence(2,4,5);
+        for(Flowers f : flowersS){
```
Comment (Aleksandr04) [inline src/itea/hmwk3/Runner.java]: refactor to Stream

File: src/itea/hmwk3/Chamomile.java (hunk 6 above)
Comment (vadimkononenko) [reply src/itea/hmwk3/Chamomile.java]: done

**Pull request merged**